### PR TITLE
Fixes issue #311 Use email buffer for DEL ‘/email/:email’ route

### PR DESCRIPTION
**Pull request by deeptibaghel**
Made changes to fix issue #311 Use email buffer for DEL ‘/email/:email’ route.
Also fixed Buffer function call to take care of deprecated format.

### CODE REVIEW

**Comment by vladikoff** [comment]
@vbudhram could you please do a quick glance at this (and maybe merge?)

@deeptibaghel 

1. I filed https://github.com/mozilla/fxa-auth-db-mysql/issues/316 there are a few remaining `Buffer()` calls in the old format that can be fixed. I think just a few left!
2. I filed an auth server issue here: https://github.com/mozilla/fxa-auth-server/issues/2333
3. I noticed that you had a question about this PR in the issue itself (https://github.com/mozilla/fxa-auth-db-mysql/issues/311), to make it easier just ask questions in the pull request. If you don't have a pull request sent yet, just create a pull request and mention that it is a work in progess with your doubt / question

**Comment by deeptibaghel** [comment]
> I noticed that you had a question about this PR in the issue itself (#311), to make it easier just ask questions in the pull request. If you don't have a pull request sent yet, just create a pull request and mention that it is a work in progress with your doubt / question

Will take care :)

File: db-server/index.js [hunk 1]
```diff
@@ -100,12 +100,12 @@ function createServer(db) {
...
   api.get('/email/:email',
     op(function (req) {
-      return db.getSecondaryEmail(Buffer(req.params.email, 'hex'))
+      return db.getSecondaryEmail(req.params.email)
     })
   )
   api.get('/email/:email/account',
     op(function (req) {
-      return db.accountRecord(Buffer(req.params.email, 'hex'))
+      return db.accountRecord(req.params.email)
```
Comment (vbudhram) [inline db-server/index.js]: You should be able to remove the `Buffer` from https://github.com/deeptibaghel/fxa-auth-db-mysql/blob/82073d71ad0187b55f7a3a6edf559f2345acbe8b/db-server/index.js#L113.

File: db-server/test/backend/remote.js [hunk 2]
```diff
@@ -192,7 +192,7 @@ module.exports = function(cfg, makeServer) {
             assert.equal(!! result[2].isPrimary, false, 'isPrimary is false on thirdEmailRecord email')
             assert.equal(!! result[2].isVerified, false, 'matches secondEmail thirdEmailRecord')
 
-            return client.delThen('/account/' + user.accountId + '/emails/' + secondEmailRecord.email)
+            return client.delThen('/account/' + user.accountId + '/emails/' + emailToHex(secondEmailRecord.email))
```
Comment (vbudhram) [inline db-server/test/backend/remote.js]: Perfect! But we will also need to update the auth-server to pass an encoded email, similar to https://github.com/mozilla/fxa-auth-server/blob/45ae7b2048e9115590611a55454627d3840119a8/lib/db.js#L1069.

Ref: https://github.com/mozilla/fxa-auth-server/issues/2334

Comment (deeptibaghel) [reply db-server/test/backend/remote.js]: Yeah sure i will complete it :)

File: db-server/index.js (hunk 1 above)
Comment (deeptibaghel) [reply db-server/index.js]: Missed somehow, done now.

**Pull request merged**